File: Recursion/RFibo.c
#include<stdio.h>
#define max 100
long table[max];
void initialize_memo() {
    for (int i = 0; i < max; i++)
        table[i] = -1;
}
long int Fibom(int n) {
    if (n == 1 || n == 2)
        return 1;
    if (table[n] != -1)
        return table[n];
    table[n] = Fibom(n - 1) + Fibom(n - 2);
    return table[n];
}
int main() {
    int n;
    printf( "Enter a number: ");
    scanf("%d", &n);
    if (n < 0 || n >= max)
       printf("Please enter a number between 0 and %d.\n", max - 1);
    initialize_memo();
    printf("%ld\n", Fibom(n));
    return 0;
}

File: Recursion/Rfibo.py
max = 100
table = [-1] * max
# def initialize():
#     global table
#     for i in range(max):
#         table[i] = -1
def fibom(n):
    global table
    if (n==1 or n==2):
        return 1
    if table[n] != -1:
        return table[n]
    table[n] = fibom(n - 1) + fibom(n - 2)
    return table[n]
n= int(input("Enter a number: "))
if (n <0 or n >= max):
    print(f'Please enter a number between 0 and {max-1}.')
# initialize()
print(f'Fibonacci of {n} is {fibom(n)}')